editor/editor_properties_array_dict.cpp

```diff
@@ -587,6 +603,28 @@ bool EditorPropertyArray::_is_drop_valid(const Dictionary &p_drag_data) const {
 		return true;
 	}
 
+	if (drop_type == "resource") {
+		Ref<Resource> res = drag_data["resource"];
+		if (!res.is_valid()) {
+			return false;
+		}
+
+		String res_type = res->get_class();
+		StringName script_class;
+		if (res->get_script()) {
+			script_class = EditorNode::get_singleton()->get_object_custom_type_name(res->get_script());
+		}
+
+		for (int j = 0; j < allowed_type.get_slice_count(","); j++) {
```

Comment:
I noticed some very bad loops in the code and opened #102630
You could change this loop to match my changes.

Reply:
I noticed that too, did not want to mess with it for fear of introducing bugs. The bad loops don't matter much, because there's usually just one item in the list anyway.

Reply:
Anyway, fixed up my new addition of the bad pattern. We'll have a merge conflict on the first one though.